uaclient/tests/test_cli_enable.py

```diff
@@ -375,120 +374,46 @@ def test_invalid_service_names(
 
 
 class TestPerformEnable:
-    @mock.patch("uaclient.cli.entitlements")
-    def test_missing_entitlement_raises_keyerror(self, m_entitlements):
+    @mock.patch("uaclient.entitlements.valid_services")
+    def test_missing_entitlement_raises_keyerror(self, m_valid_services):
         """We raise a KeyError on missing entitlements
 
         (This isn't a problem because any callers of _perform_enable should
         already have rejected invalid names.)
         """
-        m_entitlements.ENTITLEMENT_CLASS_BY_NAME = {}
-
+        m_valid_services.return_value = "entitlement"
         with pytest.raises(KeyError):
             _perform_enable("entitlement", mock.Mock())
 
-    @pytest.mark.parametrize(
-        "allow_beta, beta_call_count", ((True, 0), (False, 1))
-    )
+    @pytest.mark.parametrize("allow_beta", ((True), (False)))
     @pytest.mark.parametrize("silent_if_inapplicable", (True, False, None))
     @mock.patch("uaclient.contract.get_available_resources", return_value={})
-    @mock.patch("uaclient.cli.entitlements")
+    @mock.patch(
+        "uaclient.entitlements.is_config_value_true", return_value=False
+    )
     def test_entitlement_instantiated_and_enabled(
         self,
-        m_entitlements,
+        _m_is_config_value_true,
         _m_get_available_resources,
         silent_if_inapplicable,
         allow_beta,
-        beta_call_count,
     ):
         m_entitlement_cls = mock.Mock()
         m_cfg = mock.Mock()
-        m_user_cfg = mock.PropertyMock(return_value={})
-        type(m_cfg).cfg = m_user_cfg
-        m_is_beta = mock.PropertyMock(return_value=allow_beta)
-        type(m_entitlement_cls).is_beta = m_is_beta
 
-        m_entitlements.ENTITLEMENT_CLASS_BY_NAME = {
-            "testitlement": m_entitlement_cls
-        }
+        m_ents_dict = {"testitlement": m_entitlement_cls}
 
         kwargs = {"allow_beta": allow_beta}
         if silent_if_inapplicable is not None:
             kwargs["silent_if_inapplicable"] = silent_if_inapplicable
-        ret = _perform_enable("testitlement", m_cfg, **kwargs)
-
-        assert [
-            mock.call(m_cfg, assume_yes=False)
-        ] == m_entitlement_cls.call_args_list
-
-        m_entitlement = m_entitlement_cls.return_value
-        if silent_if_inapplicable:
-            expected_enable_call = mock.call(silent_if_inapplicable=True)
-        else:
-            expected_enable_call = mock.call(silent_if_inapplicable=False)
-        assert [expected_enable_call] == m_entitlement.enable.call_args_list
-        assert ret == m_entitlement.enable.return_value
-
-        assert 1 == m_cfg.status.call_count
-        assert 1 == m_user_cfg.call_count
-        assert beta_call_count == m_is_beta.call_count
-
-    @pytest.mark.parametrize("silent_if_inapplicable", (True, False, None))
-    @mock.patch("uaclient.cli.entitlements")
-    def test_beta_entitlement_not_enabled(
-        self, m_entitlements, silent_if_inapplicable
-    ):
```

Comment:
I think a full fix of #1588 will still need these tests right? I can add them back in my PR if you want though

Reply:
Since we are no longer handling the beta services on cli directly, those tests were behaving exactly as the tests we have for unknown services, that's why I removed. But if you think you will need them on #1588, no problem at all adding them back there.

Reply:
Very good point! Yes they can stay removed here then